deckhouse-controller/pkg/controller/deckhouse-release/check_release.go

```diff
@@ -601,6 +615,15 @@ func (dcr *DeckhouseReleaseChecker) generateChangelogForEnabledModules() map[str
 	return enabledModulesChangelog
 }
 
+func (dcr *DeckhouseReleaseChecker) listTags() ([]string, error) {
+	var err error
+	if dcr.tags == nil {
+		dcr.tags, err = dcr.registryClient.ListTags()
```

Comment:
If we have requested information from the repository once, we will have information about tags at that point in time in the structure. And since in subsequent attempts this information will no longer be nil, we will not be able to update information about tags in the future.
Here we need to either update the data on tags every time. Or do it periodically.

Reply:
I think it is not necessary now, because this object exists for a very limited time (request scoped)